packages/core/src/commands/insertContentAt.ts

```diff
@@ -140,6 +141,13 @@ export const insertContentAt: RawCommands['insertContentAt'] = (position, value,
     } else {
       newContent = content
 
+      const fromSelectionAtStart = selection.$from.parentOffset === 0
+      const isTextSelection = selection.$from.node().isText || selection.$from.node().isTextblock
+
+      if (fromSelectionAtStart && isTextSelection) {
```

Comment:
Can you also check inserting something that is nested like right before the text of a list item just to see what it does now vs what it will do with this change?

Don't need a test for it explicitly or anything, just knowing that the behavior is the same is enough for me

Reply:
In the case of doing this inside a list item, the image will be inserted as the first child of the list item followed by the paragraph. That's how our list item schema is currently setup afaik.

The same happens for blockquotes (or basically everything where paragraphs are nested into a block) and the schema allows for insertions.